src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -376,6 +376,20 @@ else if (exercise instanceof ProgrammingExercise) {
 
             long numberOfAssessments = resultService.countNumberOfFinishedAssessmentsForExercise(exercise.getId());
 
+            long numberOfComplaints = complaintRepository.countByResult_Participation_Exercise_IdAndComplaintType(exercise.getId(), ComplaintType.COMPLAINT);
+            long numberOfComplaintResponses = complaintResponseRepository.countByComplaint_Result_Participation_Exercise_Id_AndComplaint_ComplaintType(exercise.getId(),
+                    ComplaintType.COMPLAINT);
+            // TODO: Hanya, maybe complaints about your own assessment (numberOfTutorComplaints) should be taken into consideration
+            exercise.setNumberOfOpenComplaints(numberOfComplaints - numberOfComplaintResponses);
+            exercise.setNumberOfComplaints(numberOfComplaints);
+
+            long numberOfMoreFeedbackRequests = complaintRepository.countByResult_Participation_Exercise_IdAndComplaintType(exercise.getId(), ComplaintType.MORE_FEEDBACK);
+            long numberOfMoreFeedbackComplaintResponses = complaintResponseRepository.countByComplaint_Result_Participation_Exercise_Id_AndComplaint_ComplaintType(exercise.getId(),
+                    ComplaintType.MORE_FEEDBACK);
+            // TODO: Hanya, maybe feedback about your own assessment (numberOfTutorMoreFeedbackRequests) should be taken into consideration
+            exercise.setNumberOfOpenMoreFeedbackRequests(numberOfMoreFeedbackRequests - numberOfMoreFeedbackComplaintResponses);
+            exercise.setNumberOfMoreFeedbackRequests(numberOfMoreFeedbackRequests);
+
```

Comment:
This method already is too big imo. I also think we should move most of the code here into the `CourseService`, as logic like this should not be in a resource. Could you please start by just moving your code there? You also have some code duplication here, so you can just use the same method twice with a parameter `ComplaintType complaintType`.
In fact, I would move the complaint related requests to the complaint service

Reply:
@maxr96 I just saw now that you are also refactoring `CourseResource` in #1149 . Is it ok if I apply these changes suggested by @ungaralex  here or would this be conflicting with the changes you had in mind for your PR?

Reply:
@HanyaElhashemy no this will not conflict my changes. You can apply them here or we can also discuss this after developers meeting.